dsl/src/main/java/tech/pegasys/peeps/network/Network.java

```diff
@@ -387,40 +403,57 @@ private void everyMember(Consumer<NetworkMember> action) {
     members.parallelStream().forEach(action);
   }
 
-  private Genesis createGenesis(
+  private Map<Web3ProviderType, Genesis> createGenesis(
       final ConsensusMechanism consensus,
       final Map<GenesisAddress, GenesisAccount> genesisAccounts,
       final Web3Provider... validators) {
     final long chainId = Math.round(Math.random() * Long.MAX_VALUE);
 
-    final GenesisConfig genesisConfig;
-    final GenesisExtraData extraData;
-
-    switch (consensus) {
-      case CLIQUE:
-        genesisConfig = new GenesisConfigClique(chainId, new CliqueConfig());
-        extraData = new GenesisExtraDataClique(validators);
-        break;
-      case IBFT2:
-        genesisConfig = new GenesisConfigIbft2(chainId, new BftConfig());
-        extraData = new GenesisExtraDataIbft2(validators);
-        break;
-      case IBFT:
-        genesisConfig = new GenesisConfigIbftLegacy(chainId, new IbftLegacyConfig());
-        extraData = new GenesisExtraDataIbftLegacy(validators);
-        break;
-      case QBFT:
-        genesisConfig = new GenesisConfigQbft(chainId, new BftConfig());
-        extraData = new GenesisExtraDataIbft2(validators);
-        break;
-      case ETH_HASH:
-      default:
-        extraData = null;
-        genesisConfig = new GenesisConfigEthHash(chainId, new EthHashConfig());
-        break;
-    }
-
-    return new Genesis(genesisConfig, genesisAccounts, extraData);
+    final Map<Web3ProviderType, Genesis> result = new HashMap<>();
+
+    Stream.of(Web3ProviderType.values())
+        .forEach(
+            e -> {
+              final GenesisConfig genesisConfig;
+              final GenesisExtraData extraData;
+
+              switch (consensus) {
+                case CLIQUE:
+                  genesisConfig = new GenesisConfigClique(chainId, new CliqueConfig());
+                  extraData = new GenesisExtraDataClique(validators);
+                  break;
+                case IBFT2:
+                  genesisConfig = new GenesisConfigIbft2(chainId, new BftConfig());
+                  extraData = new GenesisExtraDataIbft2(validators);
+                  break;
+                case IBFT:
+                  if (e.equals(Web3ProviderType.BESU)) {
+                    genesisConfig = new BesuConfigIbft(chainId, new BesuLegacyIbftOptions());
+                  } else {
+                    genesisConfig = new GoQuorumIbftConfig(chainId, new GoQuorumIbftOptions());
+                  }
+                  extraData = new GenesisExtraDataIbftLegacy(validators);
+                  break;
+                case QBFT:
+                  if (e.equals(Web3ProviderType.BESU)) {
```

Comment:
since these are enums you can use == as well.

Reply:
done - bit cleaner.